templates/sfn/ingest_sfn_definition.json.tpl

```diff
@@ -0,0 +1,153 @@
+{
+  "Comment": "${step_function_name}: A State machine for Judgment ingests",
+  "StartAt": "Map metadata",
+  "States": {
+    "Map metadata": {
+      "Type": "Task",
+      "Resource": "arn:aws:states:::lambda:invoke",
+      "OutputPath": "$.Payload",
+      "Parameters": {
+        "Payload.$": "$",
+        "FunctionName": "arn:aws:lambda:eu-west-2:${account_id}:function:${ingest_mapper_lambda_name}"
+      },
+      "Retry": [
+        {
+          "ErrorEquals": [
+            "Lambda.ServiceException",
+            "Lambda.AWSLambdaException",
+            "Lambda.SdkClientException",
+            "Lambda.TooManyRequestsException"
+          ],
+          "IntervalSeconds": 2,
+          "MaxAttempts": 6,
+          "BackoffRate": 2
+        }
+      ],
+      "Next": "Create or update folders in Preservica"
+    },
+    "Create or update folders in Preservica": {
+      "Type": "Task",
+      "Resource": "arn:aws:states:::lambda:invoke",
+      "Parameters": {
+        "FunctionName": "arn:aws:lambda:eu-west-2:${account_id}:function:${ingest_upsert_archives_folder_lambda_name}",
+        "Payload.$": "$"
+      },
+      "Retry": [
+        {
+          "ErrorEquals": [
+            "Lambda.ServiceException",
+            "Lambda.AWSLambdaException",
+            "Lambda.SdkClientException",
+            "Lambda.TooManyRequestsException"
+          ],
+          "IntervalSeconds": 2,
+          "MaxAttempts": 6,
+          "BackoffRate": 2
+        }
+      ],
+      "Next": "Map over each Asset Id",
+      "ResultPath": null
+    },
+    "Map over each Asset Id": {
+      "Type": "Map",
+      "ItemProcessor": {
+        "ProcessorConfig": {
+          "Mode": "INLINE"
+        },
+        "StartAt": "Create Asset OPEX",
+        "States": {
+          "Create Asset OPEX": {
+            "Type": "Task",
+            "Resource": "arn:aws:states:::lambda:invoke",
+            "OutputPath": "$.Payload",
+            "Parameters": {
+              "Payload.$": "$",
+              "FunctionName": "arn:aws:lambda:eu-west-2:${account_id}:function:${ingest_asset_opex_creator_lambda_name}"
+            },
+            "Retry": [
+              {
+                "ErrorEquals": [
+                  "Lambda.ServiceException",
+                  "Lambda.AWSLambdaException",
+                  "Lambda.SdkClientException",
+                  "Lambda.TooManyRequestsException"
+                ],
+                "IntervalSeconds": 2,
+                "MaxAttempts": 6,
+                "BackoffRate": 2
+              }
+            ],
+            "End": true
+          }
+        }
+      },
+      "ItemSelector": {
+        "id.$": "$$.Map.Item.Value",
+        "batchId.$": "$$.Execution.Input.batchId",
+        "executionName.$": "$$.Execution.Name",
+        "sourceBucket.$": "$.s3Bucket"
+      },
+      "Next": "Convert array to object",
+      "ItemsPath": "$.contentAssets",
```

Comment:
Readability - can ItemsPath and ItemSelector be above the nested date definition (and in that order)? This reads like a do/while loop without the initial iteration :nauseated_face: